src/librustc_metadata/encoder.rs

```diff
@@ -541,6 +541,10 @@ impl<'tcx> EncodeContext<'tcx> {
             extra_filename: tcx.sess.opts.cg.extra_filename.clone(),
             triple: tcx.sess.opts.target_triple.clone(),
             hash: tcx.crate_hash(LOCAL_CRATE),
+            // This field is never filled when encoding metadata. Instead, it will be injected at
+            // load time when resolving dependencies. It will then be tracked in the transitive
+            // dependency information in the `CrateDep`.
```

Comment:
This means `host_hash` should be a field of `CrateMetadata` rather than `CrateRoot`.

Reply:
Otherwise LGTM.